### PR TITLE
feat: accept autocomplete attr on textbox - OKTA-326235

### PR DESCRIPTION
**Pull request by shuowu**
## Description:

Accepts `autoComplete` option in TextBox and PasswordBox components to override the `autocomplete` attribute (default as `off`) of form input.

This PR only adds autocomplete attr for `PrimaryAuthForm`.

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-326235](https://oktainc.atlassian.net/browse/OKTA-326235)

### CODE REVIEW

File: src/widget/OktaSignIn.js [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
 import Errors from 'util/Errors';
 import OAuth2Util from 'util/OAuth2Util';
 import Util from 'util/Util';
 import createAuthClient from 'widget/createAuthClient';
 import V1Router from 'LoginRouter';
 import V2Router from 'v2/WidgetRouter';
 
+// Enable "autoComplete" attribute
+const { FormUtil } = internal.views.forms.helpers;
+FormUtil.INPUT_OPTIONS.push('autoComplete');
```
Comment (haishengwu-okta) [inline src/widget/OktaSignIn.js:13]: Why not change Courage?

**Review by haishengwu-okta — CHANGES_REQUESTED** [review]

File: src/widget/OktaSignIn.js (hunk 1 above)
Comment (shuowu) [reply src/widget/OktaSignIn.js:13]: Initially, I was planning to limit the change in SIW only, but looks like the change should be safe to stay in courage. PR added here: https://github.com/okta/okta-ui/pull/3371

File: src/models/Settings.js [hunk 2]
```diff
@@ -472,4 +472,8 @@ export default Model.extend({
 
     return options;
   },
+
+  isDsTheme: function () {
```
Comment (haishengwu-okta) [inline src/models/Settings.js:476]: FYI:
BaseForm expects the `this.settings` is instance of `Courage/SettingsModel` but apparently SIW creates its own settings (`src/models/Settings`) which is incompatible with Courage's SettingsModel.

**Review by haishengwu-okta — APPROVED** [review]
🚀

**Review by aarongranick-okta — COMMENTED** [review]
I tested with the latest build `5.3.0-g8bf77f4`. I see that it is now saving and suggesting the username field! However, it is not offering to save password.

**Pull request closed without merging**